axiom/datasets.go

```diff
@@ -331,6 +333,60 @@ func (s *DatasetsService) IngestEvents(ctx context.Context, id string, opts Inge
 	return &res, nil
 }
 
+// IngestChannel ingests events from a channel into the dataset identified by
+// its id. As it keeps a connection open until the channel is closed, it is not
+// advised to use this method for long-running ingestions.
```

Comment:
By "not advised" we really mean that anyone not reading the docs here can easily DOS us I think? https://linear.app/axiom/issue/DB-3420/drip-feed-data-into-ingest-results-in-super-high-ingest-pressure-dos

I am a little worried exposing this as public API in our current state TBH. I think it is a nice API, but it is extremely easy to misuse unknowingly and cause us great trouble (until we fix the linked issue).